### PR TITLE
Remove extraneous function prototypes from c.h

### LINKED ISSUE #1324: Functions were removed but prototypes weren't
In commit 3214905 the following functions were removed: `rocksdb_options_set_max_grandparent_overlap_factor`, `rocksdb_options_set_expanded_compaction_factor`, and `rocksdb_options_set_source_compaction_factor`.

However in `include/rocksdb/c.h` those functions still have prototypes on lines [565](https://github.com/facebook/rocksdb/blob/master/include/rocksdb/c.h#L565), [562](https://github.com/facebook/rocksdb/blob/master/include/rocksdb/c.h#L562), and [661](https://github.com/facebook/rocksdb/blob/master/include/rocksdb/c.h#L661). If somebody attempted to use those functions, their program would compile but not link.

### PR DESCRIPTION
**Pull request by ammongit**
This fixes #1324.

### CODE REVIEW

**Comment by facebook-github-bot-8** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

**Comment by ammongit** [comment]
I already went to that link and signed the CLA. Is there anything else I need to do?

**Comment by facebook-github-bot-5** [comment]
Thank you for signing our Contributor License Agreement. We can now accept your code for this (and any) Facebook open source project. Thanks!

File: include/rocksdb/c.h [hunk 1]
```diff
@@ -662,8 +658,8 @@ extern ROCKSDB_LIBRARY_API void rocksdb_options_set_source_compaction_factor(
     rocksdb_options_t*, int);
 extern ROCKSDB_LIBRARY_API void rocksdb_options_prepare_for_bulk_load(
     rocksdb_options_t*);
-extern ROCKSDB_LIBRARY_API void rocksdb_options_set_memtable_vector_rep(
```
Comment (ajkr) [inline include/rocksdb/c.h]: is this one still needed? https://github.com/facebook/rocksdb/blob/master/db/c.cc#L1780

Comment (ammongit) [inline include/rocksdb/c.h]: Good catch, that's not the right function. I was supposed to delete line 661.

**Comment by ajkr** [comment]
LGTM, thank you!

**Pull request merged**